src/mesh/mesh.cc

```diff
@@ -773,5 +739,48 @@ const BIHTree &Mesh::get_bih_tree() {
     return *bih_tree_;
 }
 
+
+void Mesh::add_physical_name(unsigned int dim, unsigned int id, std::string name) {
+	region_db_.add_region(id, name, dim, "$PhysicalNames");
+}
+
+
+void Mesh::add_node(unsigned int node_id, arma::vec3 coords) {
+	NodeFullIter node = node_vector.add_item(node_id);
+	node->point() = coords;
+}
+
+
+void Mesh::add_element(unsigned int elm_id, unsigned int dim, unsigned int region_id, unsigned int partition_id,
+		std::vector<unsigned int> node_ids) {
+	Element *ele=nullptr;
+	RegionIdx region_idx = region_db_.get_region( region_id, dim );
+	if ( !region_idx.is_valid() ) {
+		region_idx = region_db_.add_region( region_id, region_db_.create_label_from_id(region_id), dim, "$Element" );
+	}
+	region_db_.mark_used_region(region_idx.idx());
+
+	if (region_idx.is_boundary()) {
+		ele = bc_elements.add_item(elm_id);
+	} else {
+		if(dim == 0 )
+			WarningOut().fmt("Bulk elements of zero size(dim=0) are not supported. Element ID: {}.\n", elm_id);
```

Comment:
We are potential problem here. If warning is produced, 'ele' object is not initialized and next method 'ele-&gt;init' failed on segfault.